docs: automatic `always` tag in gather_facts

yajo opened the pull request:
Since https://github.com/ansible/ansible/pull/44717 got merged, Ansible adds the `always` tag to the gather_facts plugin included in plays only if the play has no further tags.

This behavior makes sense, but is surprising if not documented.

@moduon MT-2305

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

mkrizek commented:
FWIW this is already documented, from https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_tags.html#special-tags-always-and-never:

> Fact gathering is tagged with ‘always’ by default. It is only skipped if you apply a tag and then use a different tag in --tags or the same tag in --skip-tags.

yajo committed:
docs: automatic `always` tag in gather_facts

Since https://github.com/ansible/ansible/pull/44717 got merged, Ansible adds the `always` tag to the gather_facts plugin included in plays only if the play has no further tags.

This behavior makes sense, but is surprising if not documented.

moduon MT-2305

yajo commented:
OK I see, I skipped that part... Well, I amended it a bit to be more clear on where the tag is applied. Thanks!

samccann commented:
/azp run

samccann commented:
@mkrizek can you take another quick look? I'm weak on tags etc and this makes the note specific to plays, but the text above says the tag can be  at the play or task level...

mkrizek commented:
> @mkrizek can you take another quick look? I'm weak on tags etc and this makes the note specific to plays, but the text above says the tag can be at the play or task level...

The `gather_facts` is a play keyword and as such only tags specified on a play have an effect on it. So when a play is tagged and tags on the command line are specified (`--tags`/`--skip-tags`) in a way that the whole play is skipped, the fact gathering will be skipped as well (since it is a part of the play).

mkrizek commented:
And I should also note that for a case when a play is not tagged, the gathering facts task (the `setup` module) is tagged with the `always` tag by default so tags specified on the command line don't have an effect on it in such a case.

Hope that explains the warning message.

samccann commented:
@yajo thanks for the Ansible docs fix! And thanks for the feedback and explanation @mkrizek !!
